lib/utilities/firebase_service.dart

```diff
@@ -108,108 +130,274 @@ class FirebaseService {
+        }
+      }
+    }
+    return userNames;
+  } catch (e) {
+    print('Error fetching user names: $e');
+    return [];
+  }
+}
+  /// Uploads a profile picture to Firebase Storage and updates the Firestore user document
+Future<void> uploadProfilePicture(String userId, String filePath) async {
+  try {
+    final File imageFile = File(filePath);
+
+    // Reference to the storage location
+    final storageRef = _storage.ref().child('profile_pictures/$userId.jpg');
+
+    // Upload the file to Firebase Storage
+    await storageRef.putFile(imageFile);
+
+    // Get the download URL of the uploaded file
+    final String downloadUrl = await storageRef.getDownloadURL();
+
+    // Update the user document in Firestore with the download URL
+    await _firestore.collection('users').doc(userId).update({
+      'profilePicture': downloadUrl,
+    });
+
+    print('Profile picture uploaded successfully: $downloadUrl');
+  } catch (e) {
+    print('Error uploading profile picture: $e');
+    throw e;
+  }
+}
+
+
+  /// Deletes the profile picture from Firebase Storage and removes it from Firestore
+  Future<void> deleteProfilePicture(String userId) async {
+    try {
+      // Reference to the storage location
+      final storageRef = _storage.ref().child('profile_pictures/$userId.jpg');
+
+      // Delete the image from Firebase Storage
+      await storageRef.delete();
+
+      // Remove the profilePicture field from the Firestore user document
+      await _firestore.collection('users').doc(userId).update({
+        'profilePicture': FieldValue.delete(),
+      });
+
+      print('Profile picture deleted successfully');
+    } catch (e) {
+      print('Error deleting profile picture: $e');
+    }
+  }
+  Future<DocumentSnapshot<Map<String, dynamic>>> _getUserDocument(String userId) {
+  return _firestore.collection('users').doc(userId).get();
+}
+
+  /// Fetches the current user's profile picture URL
+  Future<String?> getProfilePicture(String userId) async {
+  try {
+    final userDoc = await _getUserDocument(userId);
+    if (userDoc.exists) {
+      final data = userDoc.data(); // Safely retrieve data
+      if (data != null && data['profilePicture'] != null) {
+        return data['profilePicture'] as String; // Return profile picture URL
+      }
+    }
+    return null; // Return null if no profile picture is found
+  } catch (e) {
+    print('Error fetching profile picture: $e');
+    return null; // Return null in case of an error
+  }
+}
+// Update user details
+  Future<void> updateUserProfile(String userId, String name, String email) async {
+    await _firestore.collection('users').doc(userId).update({
+      'name': name,
+      'email': email,
+    });
+  }
+
+  // Delete user account
+  Future<void> deleteAccount(String userId) async {
+    // Delete user document
+    await _firestore.collection('users').doc(userId).delete();
+
+    // Delete user authentication
+    await FirebaseAuth.instance.currentUser?.delete();
```

Comment:
You should have injected Firebase auth and not use the static instance. It's the dependency injection pattern that we should have followed.